gridcomps/History/MAPL_StationSamplerMod.F90

```diff
@@ -0,0 +1,461 @@
+           !!   call this%vdata%regrid_eta_to_pressure(p_src_3d,p_new_lev,_RC)
+           !!   call this%regridder%regrid(p_new_lev,p_dst_3d,_RC)
+           !!else
+           call this%regridder%regrid(p_src_3d,p_dst_3d,_RC)
+           !!end if
+           !! 
+           if (mapl_am_i_root()) then
+              nx=size(p_dst_3d,1); nz=size(p_dst_3d,2); allocate(arr(nz, nx))
+              arr=reshape(p_dst_3d,[nz,nx],order=[2,1])
+              call this%formatter%put_var(xname,arr,&
+                   start=[1,1,this%obs_written],count=[nz,nx,1],_RC)
+              !note:     lev,station,time
+           end if
+           call ESMF_FieldDestroy(dst_field,nogarbage=.true.)
+        else
+           _FAIL('grid2LS regridder: rank > 3 not implemented')
+        end if  ! rank
+     enddo      ! fieldCount
+     _RETURN(_SUCCESS)
+   end subroutine append_file
+
+   
+   subroutine create_file_handle(this,filename,rc)
+     class(StationSampler), intent(inout) :: this
+     character(len=*), intent(inout) :: filename  ! for ouput nc
+     integer, optional, intent(out) :: rc
+     type(variable) :: v
+     integer :: status
+
+     this%ofile = trim(filename)
+     v = this%time_info%define_time_variable(_RC)
+     call this%fmd%modify_variable('time',v,_RC)
+     !
+     if (mapl_am_I_root()) then
+        call this%formatter%create(trim(filename),_RC)
+        call this%formatter%write(this%fmd,_RC)
+        call this%formatter%put_var('longitude',this%lons,_RC)
+        call this%formatter%put_var('latitude',this%lats,_RC)
+        call this%formatter%put_var('station_id',this%station_id,_RC)                
+     end if
+     this%obs_written = 0
+     _RETURN(_SUCCESS)
+   end subroutine create_file_handle
+
+  
+   subroutine close_file_handle(this,rc)
+     class(StationSampler), intent(inout) :: this
+     integer, optional, intent(out) :: rc
+     integer :: status
+     if (trim(this%ofile) /= '') then
+        if (mapl_am_i_root()) then
+           call this%formatter%close(_RC)
+        end if
+     end if
+     _RETURN(_SUCCESS)
+   end subroutine close_file_handle
+   
+
+   function compute_time_for_current(this,current_time,rc) result(rtimes)
+     class(StationSampler), intent(inout) :: this
+     type(ESMF_Time), intent(in) :: current_time
+     integer, optional, intent(out) :: rc
+     real(ESMF_KIND_R8), allocatable :: rtimes(:)
+     integer :: i,status
+     type(ESMF_TimeInterval) :: tint
+     type(ESMF_Time) :: file_start_time
+     character(len=ESMF_MAXSTR) :: tunits
+     
+     allocate(rtimes(1),_STAT)
+     call this%get_file_start_time(file_start_time,tunits,_RC)
+     tint = current_time-file_start_time
+     select case(trim(tunits))
+     case ('days')
+        call ESMF_TimeIntervalGet(tint,d_r8=rtimes(1),_RC)
+     case ('hours')
+        call ESMF_TimeIntervalGet(tint,h_r8=rtimes(1),_RC)
+     case ('minutes')
+        call ESMF_TimeIntervalGet(tint,m_r8=rtimes(1),_RC)
+     end select
```

Comment:
```suggestion
     case default
        _FAIL('illegal value for tunits: ' // tunits)
     end select
```
Maybe this is not necessary - can we prove that `get_file_start_time` will always return a valid `tunits`?

Reply:
This time process mechanism comes from 
https://github.com/GEOS-ESM/MAPL/blob/main/base/MAPL_TimeMethods.F90#L88-L106

[MAPL_TimeMethods.F90
 type, public :: timeData
function define_time_variable(this,rc) result(v)]

The constructor is instantiated in MAPL_HistoryGridComp.F90
https://github.com/GEOS-ESM/MAPL/blob/main/gridcomps/History/MAPL_HistoryGridComp.F90#L2345-L2352
e.g.,
 list(n)%timeInfo = TimeData(clock,tm,MAPL_nsecf(list(n)%frequency),IntState%stampoffset(n),integer_time=intstate%integer_time)

Hence, the function compute_time_for_current here is defined on previous layout.